py/foundation/entity.py

```diff
@@ -48,6 +48,9 @@ def to_proto(self) -> PbEntityPayloadType:
     """
     ...
 
+  def text(self) -> Optional[str]:
```

Comment:
I am kind of confused about why an Entity would have text. What if it is an entity like a checkbox?

Reply:
Yeah my thinking was that it could be useful for handling a lot of potentially repetitive boilerplate when we want to know what the text in a line or paragraph is, basically just rendering lines by putting spaces between words and newlines between lines. Seems like it could come in handy, and along the lines of being like `__str__`, except it might also return `None`. If it's something like a checkbox, it would return `None`.

Of course, this may just not be something we need on Entity at all. If this would be entirely BP-specific, we could probably just add that functionality in BP for entities.